src/ae_kqueue.c

```diff
@@ -36,8 +36,29 @@
 typedef struct aeApiState {
     int kqfd;
     struct kevent *events;
+
+    /* Events mask for merge read and write event.
+     * To reduce memory consumption, we use 2 bits to store the mask
+     * of an event, so that 1 byte will store the mask of 4 events. */
+    char *eventsMask; 
 } aeApiState;
 
+#define EVENT_MASK_MALLOC_SIZE(sz) (((sz) + 3) / 4)
+#define EVENT_MASK_OFFSET(fd) (6 - (fd) % 4 * 2)
```

Comment:
why we we need to inverse it? we have a value from 0..6, i don't see any reason to translate it to 6..0
```suggestion
#define EVENT_MASK_OFFSET(fd) ((fd) % 4 * 2)
```

Reply:
I'm not sure I'm getting your point right.

fd0 -> 01
fd5 -> 10
Then the binary of fd0-fd5 should be 01xxxxxx10.
`EVENT_MASK_OFFSET` calculates the fd mask and the rightmost offset, if we want to update `fd0` to `11`, we need to put `0x3 << 6`, which is the position we want to update.

Reply:
Are you trying to store in the opposite way?
fd0 -> 01
fd5 -> 10
Then the binary of fd0-fd5 should be 10xxxxxx01.